### PR TITLE
Enable ingester to querier chunks streaming and ingester query request minimisation by default

### PR DESCRIPTION
**Pull request by charleskorn**
#### What this PR does

This PR enables two recently-introduced features by default, and marks them as no longer being experimental:

* ingester to querier chunks streaming (store-gateway to querier chunks streaming will remain experimental for now)
* ingester query request minimisation

Both of these features have been enabled in production at Grafana Labs for quite some time now, and we've had no issues with either of them.

#### Which issue(s) this PR fixes or relates to

(none)

#### Checklist

- [n/a] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by charleskorn** [comment]
Putting this back to draft while I investigate an issue I've just noticed on an internal Grafana Labs cluster.

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,8 @@
 * [CHANGE] General: enabled `-log.buffered` by default. The `-log.buffered` has been deprecated and will be removed in Mimir 2.13. #6131
 * [CHANGE] Ingester: changed default `-blocks-storage.tsdb.series-hash-cache-max-size-bytes` setting from `1GB` to `350MB`. The new default cache size is enough to store the hashes for all series in a ingester, assuming up to 2M in-memory series per ingester and using the default 13h retention period for local TSDB blocks in the ingesters. #6129
 * [CHANGE] Query-frontend: removed `cortex_query_frontend_workers_enqueued_requests_total`. Use `cortex_query_frontend_enqueue_duration_seconds_count` instead. #6121
+* [CHANGE] Ingester / querier: enable ingester to querier chunks streaming by default and mark it as not experimental. #6174
+* [CHANGE] Ingester / querier: enable ingester query request minimisation by default and mark it as not experimental. #6174
```
Comment (pracucci) [inline CHANGELOG.md:13]: [nit] I would go straight to the point ;)

```suggestion
* [CHANGE] Ingester / querier: enable ingester to querier chunks streaming by default and mark it as stable. #6174
* [CHANGE] Ingester / querier: enable ingester query request minimisation by default and mark it as stable. #6174
```

**Review by pracucci — APPROVED** [review]
I left you investigate the issue you've mentioned. If turns out it was a false positive, then I'm happy to mark it as stable and enable it by default. Good job!

**Review by dimitarvdimitrov — COMMENTED** [review]
are there cases where this feature has to be disabled? I'm wondering if it's not a bad idea to enable them by default but then remove the flag after some time and always have the feature enabled?

**Comment by charleskorn** [comment]
>are there cases where this feature has to be disabled? I'm wondering if it's not a bad idea to enable them by default but then remove the flag after some time and always have the feature enabled? That will help keep the configuration surface of mimir smaller

Agree. There are no cases where you'd want to disable this based on our testing. I'd like to leave the flag there for a little while just in case there's an issue we haven't already run into, but I'd absolutely like to make streaming the default and remove the flag altogether if things go smoothly.

**Comment by dimitarvdimitrov** [comment]
> > are there cases where this feature has to be disabled? I'm wondering if it's not a bad idea to enable them by default but then remove the flag after some time and always have the feature enabled? That will help keep the configuration surface of mimir smaller
> 
> Agree. There are no cases where you'd want to disable this based on our testing. I'd like to leave the flag there for a little while just in case there's an issue we haven't already run into, but I'd absolutely like to make streaming the default and remove the flag altogether if things go smoothly.

my comment was mostly about keeping the flag as experimental in this PR so that we can remove it without going through the deprecation process

**Comment by charleskorn** [comment]
> my comment was mostly about keeping the flag as experimental in this PR so that we can remove it without going through the deprecation process

Right, makes sense. Sounds good to me, I'll switch it back to experimental.

**Comment by charleskorn** [comment]
https://github.com/grafana/mimir/pull/6346 fixed the issue I discovered and we've had no instances of the issue since rolling out that fix, so I believe this is good to go again.

**Pull request merged**